__tests__/plugins/rabbitmq_plugin.test.js

```diff
@@ -8,32 +8,110 @@
 import { bigIntUtils } from '@hathor/wallet-lib';
 import { eventHandlerFactory, getSettings } from '../../src/plugins/hathor_rabbitmq';
 
-test('settings', () => {
-  const oldArgs = process.argv;
-  process.argv = [
-    'node', // not used but a value is required at this index
-    'a_script_file.js', // not used but a value is required at this index
-    '--plugin_rabbitmq_url', 'test-url',
-    '--plugin_rabbitmq_queue', 'test-queue',
-  ];
-  const settings = getSettings();
-  expect(settings).toMatchObject({
-    url: 'test-url',
-    queue: 'test-queue',
-  });
-
-  // Restore original argv state
-  process.argv = oldArgs;
+describe('RabbitMQ plugin settings', () => {
+  it('should throw an error if no settings are provided', () => {
+    const oldArgs = process.argv;
+    process.argv = [
+      'node', // not used but a value is required at this index
+      'a_script_file.js', // not used but a value is required at this index
+    ];
+    expect(() => getSettings()).toThrow('You must provide a RabbitMQ URL');
+    process.argv = oldArgs;
+  });
+
+  it('should throw an error if no queue or exchange is provided', () => {
+    const oldArgs = process.argv;
+    process.argv = [
+      'node', // not used but a value is required at this index
+      'a_script_file.js', // not used but a value is required at this index
+      '--plugin_rabbitmq_url', 'test-url',
+    ];
+    expect(() => getSettings()).toThrow('You must provide either a RabbitMQ queue or exchange');
+    process.argv = oldArgs;
+  });
+
+  it('should throw an error if both queue and exchange are provided', () => {
+    const oldArgs = process.argv;
+    process.argv = [
+      'node', // not used but a value is required at this index
+      'a_script_file.js', // not used but a value is required at this index
+      '--plugin_rabbitmq_url', 'test-url',
+      '--plugin_rabbitmq_queue', 'test-queue',
+      '--plugin_rabbitmq_exchange', 'test-exchange',
+    ];
+    expect(() => getSettings()).toThrow('You must provide either a RabbitMQ queue or exchange, not both');
+    process.argv = oldArgs;
+  });
+
+  it('should throw an error if exchange is provided without routing key', () => {
+    const oldArgs = process.argv;
+    process.argv = [
+      'node', // not used but a value is required at this index
+      'a_script_file.js', // not used but a value is required at this index
+      '--plugin_rabbitmq_url', 'test-url',
+      '--plugin_rabbitmq_exchange', 'test-exchange',
+    ];
+    expect(() => getSettings()).toThrow('You must provide a RabbitMQ routing key if you provide exchange. A blank routing key is acceptable though.');
```

Comment:
suggestion(non-blocking): Not related to this line, but I think we should have a test for this edge case. An exchange with a blank routing key.

This will ensure the command line parser is working as expected. This doesn't need to be implemented in this PR, however, as it's just an improvement upon this working set of tests.

Reply:
Done in https://github.com/HathorNetwork/hathor-wallet-headless/pull/531/commits/0301b03f9fb11d0c2e28adc9b539e9d5a93218cc